src/analysis/fourier/fourier_add_circle.lean

```diff
@@ -0,0 +1,330 @@
+  ext1 x, apply fourier_neg
+end
+
+variables [hT : fact (0 < T)]
+include hT
+
+/-- The subalgebra of `C(add_circle T, ℂ)` generated by `fourier n` for `n ∈ ℤ`
+separates points. -/
+lemma fourier_subalgebra_separates_points : (@fourier_subalgebra T).separates_points :=
+begin
+  intros x y hxy,
+  refine ⟨_, ⟨fourier 1, subset_adjoin ⟨1, rfl⟩, rfl⟩, _⟩,
+  dsimp only, rw [fourier_one, fourier_one],
+  contrapose! hxy,
+  rw subtype.coe_inj at hxy,
+  exact injective_to_circle hT.elim.ne' hxy,
+end
+
+/-- The subalgebra of `C(add_circle T, ℂ)` generated by `fourier n` for `n ∈ ℤ` is dense. -/
+lemma fourier_subalgebra_closure_eq_top : (@fourier_subalgebra T).topological_closure = ⊤ :=
+continuous_map.subalgebra_is_R_or_C_topological_closure_eq_top_of_separates_points
+  fourier_subalgebra fourier_subalgebra_separates_points fourier_subalgebra_conj_invariant
+
+/-- The linear span of the monomials `fourier n` is dense in `C(add_circle T, ℂ)`. -/
+lemma span_fourier_closure_eq_top : (span ℂ (range $ @fourier T)).topological_closure = ⊤ :=
+begin
+  rw ← fourier_subalgebra_coe,
+  exact congr_arg subalgebra.to_submodule fourier_subalgebra_closure_eq_top,
+end
+
+/-- The family of monomials `fourier n`, parametrized by `n : ℤ` and considered as
+elements of the `Lp` space of functions `add_circle T → ℂ`. -/
+abbreviation fourier_Lp (p : ℝ≥0∞) [fact (1 ≤ p)] (n : ℤ) : Lp ℂ p (@haar_add_circle T hT) :=
+to_Lp p haar_add_circle ℂ (fourier n)
+
+lemma coe_fn_fourier_Lp (p : ℝ≥0∞) [fact (1 ≤ p)] (n : ℤ) :
+  (@fourier_Lp T hT p _ n) =ᵐ[haar_add_circle] fourier n := coe_fn_to_Lp haar_add_circle (fourier n)
+
+/-- For each `1 ≤ p < ∞`, the linear span of the monomials `fourier n` is dense in
+`Lp ℂ p haar_circle`. -/
+lemma span_fourier_Lp_closure_eq_top {p : ℝ≥0∞} [fact (1 ≤ p)] (hp : p ≠ ∞) :
+  (span ℂ (range (@fourier_Lp T _ p _))).topological_closure = ⊤ :=
+begin
+  convert (continuous_map.to_Lp_dense_range ℂ hp (@haar_add_circle T hT) ℂ
+    ).topological_closure_map_submodule (span_fourier_closure_eq_top),
+  rw [map_span, range_comp],
+  simp only [continuous_linear_map.coe_coe],
+end
+
+/-- The monomials `fourier n` are an orthonormal set with respect to normalised Haar measure. -/
+lemma orthonormal_fourier : orthonormal ℂ (@fourier_Lp T _ 2 _) :=
+begin
+  rw orthonormal_iff_ite,
+  intros i j,
+  rw continuous_map.inner_to_Lp (@haar_add_circle T hT) (fourier i) (fourier j),
+  simp_rw [←fourier_neg, ←fourier_add],
+  split_ifs,
+  { simp_rw [h, neg_add_self],
+    have : ⇑(@fourier T 0) = (λ x, 1 : (add_circle T) → ℂ),
+    { ext1, exact fourier_zero },
+    rw [this, integral_const, measure_univ, ennreal.one_to_real, complex.real_smul,
+      complex.of_real_one, mul_one] },
+  have hij : -i + j ≠ 0,
+  { rw add_comm,
+    exact sub_ne_zero.mpr (ne.symm h) },
+  convert integral_eq_zero_of_add_right_eq_neg (fourier_add_half_inv_index hij hT.elim),
+  rw haar_add_circle,
+  exact is_mul_left_invariant.is_add_right_invariant
```

Comment:
Clearly `to_additive` made a mistake here with the auto-naming!  If you are feeling like a good citizen, you can fix it by adding the correct name to the `to_additive` attribute in that declaration (back in the measure theory library), something like
```lean
@[priority 100, to_additive is_add_left_invariant.is_add_right_invariant "In an abelian ..."]
```